src/backend/x11/window_inner.rs

```diff
@@ -0,0 +1,238 @@
+    cursor_visible: Arc<AtomicBool>,
+    pub depth: Depth,
+    pub gc: x11::Gcontext,
+}
+
+impl WindowInner {
+    pub fn new(
+        connection: Weak<RustConnection>,
+        screen: &Screen,
+        properties: WindowProperties<'_>,
+        atoms: Atoms,
+        depth: Depth,
+        visual_id: u32,
+        colormap: u32,
+    ) -> Result<WindowInner, X11Error> {
+        let weak = connection;
+        let connection = weak.upgrade().unwrap();
+
+        // Generate the xid for the window
+        let window = connection.generate_id()?;
+
+        // The event mask never include `EventMask::RESIZE_REDIRECT`.
+        //
+        // The reason is twofold:
+        // - We are not a window manager
+        // - Makes our window impossible to resize.
+        //
+        // On the resizing aspect, KWin and some other WMs would allow resizing, but those
+        // compositors rely on putting this window in another window for drawing decorations,
+        // so visibly in KWin it would look like using the RESIZE_REDIRECT event mask would work,
+        // but a tiling window manager would be sad and the tiling window manager devs mad because
+        // this window would refuse to listen to the tiling WM.
+        //
+        // For resizing we use ConfigureNotify events from the STRUCTURE_NOTIFY event mask.
+
+        let window_aux = CreateWindowAux::new()
+            .event_mask(
+                EventMask::EXPOSURE // Be told when the window is exposed
+            | EventMask::STRUCTURE_NOTIFY
+            | EventMask::KEY_PRESS // Key press and release
+            | EventMask::KEY_RELEASE
+            | EventMask::BUTTON_PRESS // Mouse button press and release
+            | EventMask::BUTTON_RELEASE
+            | EventMask::POINTER_MOTION // Mouse movement
+            | EventMask::NO_EVENT,
+            )
+            // Border pixel and color map need to be set if our depth may differ from the root depth.
+            .border_pixel(0)
+            .colormap(colormap);
+
+        let _ = connection.create_window(
+            depth.depth,
+            window,
+            screen.root,
+            0,
+            0,
+            properties.size.w,
+            properties.size.h,
+            0,
+            WindowClass::INPUT_OUTPUT,
+            visual_id,
+            &window_aux,
+        )?;
+
+        // Send requests to change window properties while we wait for the window creation request to complete.
+        let mut window = WindowInner {
+            connection: weak,
+            id: window,
+            root: screen.root,
+            atoms,
+            cursor_visible: Arc::new(AtomicBool::new(true)),
+            size: Mutex::new((properties.size.w, properties.size.h).into()),
+            depth,
+            gc: 0,
+        };
+
+        let gc = connection.generate_id()?;
+        connection.create_gc(gc, window.id, &CreateGCAux::new())?;
+        window.gc = gc;
```

Comment:
Instead of writing `gc: 0` above, you could just do `gc: connection.generate_id()?`. That would save like two lines of code.